harvester/src/supabase_utils.py

```diff
@@ -0,0 +1,40 @@
+import requests
+import logging
+
+
+# Function to check if a file exists in Supabase storage
+def check_file_exists_in_supabase_storage(supabaseUrl, supabaseBucketName, file_name):
+    url = (
+        f"{supabaseUrl}/storage/v1/object/info/public/{supabaseBucketName}/{file_name}"
+    )
+    response = requests.get(url)
+    return response.status_code == 200
+
+
+# Function to upload a file to Supabase storage
+def upload_file_to_supabase_storage(
+    supabaseUrl, supabaseBucketName, supabaseServiceRoleKey, file_path, file_name
+):
+    with open(file_path, "rb") as file:
+        file_url = f"{supabaseUrl}/storage/v1/object/{supabaseBucketName}/{file_name}"
+        http_method = (
+            requests.put
+            if check_file_exists_in_supabase_storage(
+                supabaseUrl, supabaseBucketName, file_name
+            )
+            else requests.post
+        )
+        response = http_method(
+            file_url,
+            files={"file": file},
+            headers={
```

Comment:
Dies this work? Isn't the `apikey` header missing? See https://supabase.com/docs/guides/api/quickstart

```
curl 'https://<PROJECT_REF>.supabase.co/rest/v1/todos' \
-H "apikey: <ANON_KEY>" \
-H "Authorization: Bearer <ANON_KEY>"
```

Reply:
But it also needs the `apikey` header

Reply:
Wired. Actually it should not.

Reply:
Tested it. Works in local instance but not with remote.

Reply:
(for future reference) As discussed, this works without api key:

```
curl --location 'https://<supbase_project_id>.supabase.co/storage/v1/object/data_assets/test.txt' \
--header 'Authorization: Bearer service_role_key' \
--form '=@"/path/to/file"'
```